--- dsdt.dsl	2017-08-15 22:05:47.316300389 +0100
+++ dsdt.fixed	2017-08-15 22:07:40.847798734 +0100
@@ -3810,6 +3810,7 @@
                         {
                             Local0 = IOST /* \IOST */
                         }
+                        Return (Zero)
                     }
 
                     Method (DCNT, 2, NotSerialized)
@@ -4153,6 +4154,7 @@
                         ^SIO1.OPT2 &= 0xFE /* \_SB_.PCI0.LPCB.SIO1.OPT2 */
                         ^SIO1.EXFG ()
                     }
+                    Return (Zero)
                 }
 
                 Method (SIOW, 1, NotSerialized)
